src/components/TaskCard.js

```diff
@@ -0,0 +1,57 @@
+import { useState } from "react";
+import Card from "@mui/material/Card";
+import CardContent from "@mui/material/CardContent";
+import Typography from "@mui/material/Typography";
+import CardActions from "@mui/material/CardActions";
+import IconButton from "@mui/material/IconButton";
+import CheckIcon from "@material-ui/icons/Check";
+import DeleteIcon from "@material-ui/icons/Delete";
+import EditIcon from "@material-ui/icons/Edit";
+import { green } from "@mui/material/colors";
+
+function TaskCard({ task, onDelete, onEdit, onCompleteChange }) {
+  const [completed, setCompleted] = useState(false);
```

Comment:
redundant state.  
You can infer the current state from the task passed through props.